pkg/util/util_test.go

```diff
@@ -53,6 +53,63 @@ func TestUntilReturnsImmediately(t *testing.T) {
 	}
 }
 
+func TestJitterUntil(t *testing.T) {
+	ch := make(chan struct{})
+	close(ch)
```

Comment:
The logic of closing the channel before the call then making a new channel is rather confusing, could you comment the logic please.